shared/app/tab-bar/index.render.js.flow

```diff
@@ -1,12 +1,13 @@
 // @flow
 import React, {Component} from 'react'
 import type {Tab} from '../../constants/tabs'
+import type {Map} from 'immutable'
 
 export type Props = {
   onTabClick: (tab: Tab) => void,
   selectedTab: Tab,
   username?: string,
-  badgeNumbers: {[key: Tab]: number},
+  badgeNumbers: Map<Tab, number>,
```

Comment:
I think in general we've been passing in plain ol' objs, and trying to keep immutable away from components.